include/itkMaskedAssignImageFilter.h

```diff
@@ -0,0 +1,124 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkMaskedAssignImageFilter_h
+#define itkMaskedAssignImageFilter_h
+
+#include "itkTernaryGeneratorImageFilter.h"
+#include "itkNumericTraits.h"
+#include "itkVariableLengthVector.h"
+#include "itkMath.h"
+
+namespace itk
+{
+
+/**
+ *\class MaskedAssignImageFilter
+ * \brief Assign values to a masked region.
+ *
+ * This class is templated over the types of the
+ * input image type, the mask image type and the type of the output image.
+ * Numeric conversions (castings) are done by the C++ defaults.
+ *
+ * The first input is the "input image", the second is the "mask image",
+ * the third is the "assign image. Any of the images can be set as constants,
+ * with third input can be set as constant with the `SetAssignConstant` method.
+ *
+ * For the "masked input" non-zero values are considered the mask, and assigned values
+ * from the "assign input".
+ *
+ * The following is the logic applied per pixel (pixel value or constant value) :
+ *
+   \code
+          if pixel_from_mask_image != 0
```

Comment:
Based on this code the mask indicates where to keep the values. I expected the mask to be the locations where the user wanted to change values, which seems to be what the implementation is doing (unless I misunderstood something in the implementation).